vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -175,7 +188,7 @@ def refresh_infrastructure_cache(self):
                 # Hosts are not considered as parents of the VMs they run, we use the `runtime.host` property
                 # to get the name of the ESXi host
                 runtime_host = properties.get("runtime.host")
-                runtime_host_props = infrastructure_data.get(runtime_host, {})
+                runtime_host_props = infrastructure_data.get(runtime_host, {})  # type: ignore
```

Comment:
"ignore" removed, it's actually not needed.

Reply:
This will lead to another error:

```
datadog_checks/vsphere/vsphere.py:189:109: error: Invalid index type "Optional[Type[ManagedEntity]]" for "Dict[ManagedEntity, InfrastructureDataItem]"; expected type "ManagedEntity"
```

Reply:
Fixing `'runtime.host': vim.ManagedEntity,` fixed this error above.